CW-331-Add-option-to-disable-market-place-in-display-settings

Blazebrain opened the pull request:
Issue Number (if Applicable): Fixes #

# Description
Add switcher in display settings to enable/disable marketplace page

if enabled: it will be the same behavior as it is right now

if disabled: the dashboard screen will only have the balance and transactions screen

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed
- [] Double check modified code and verify it with the feature/task requirements
- [x] Formate code
- [x] Look for code duplication
- [x] Clear naming for variables and methods

OmarHatem28 reviewed (changes requested):
also, you haven't added a reaction for `shouldShowMarketPlaceInDashboard` to save its new value in the shared prefs.
and please revert the formatting in `settings_store` and in `dashboard_page`

Blazebrain commented on lib/store/settings_store.dart:
```diff
@@ -232,62 +238,74 @@ abstract class SettingsStoreBase with Store {
+        false;
+    final shouldShowMarketPlaceInDashboard = sharedPreferences
+            .getBool(PreferencesKey.shouldShowMarketPlaceInDashboard) ??
+        false;
```
Fixed this  @OmarHatem28

Blazebrain commented:
> also, you haven't added a reaction for `shouldShowMarketPlaceInDashboard` to save its new value in the shared prefs to be persistent.
> and please revert the formatting in `settings_store` and in `dashboard_page`

Added the persistence also, and remove trailing commas format I added in the store and dashboard

OmarHatem28 commented:
Reviewed ok, please check Justin's comment on Jira

SamsungGalaxyPlayer commented:
@OmarHatem28 tests ok now
